### PR TITLE
added unicode support during the IPA extraction

### PR DESCRIPTION
**Pull request by Fabiano1107**
Dear All, I'm a security anlyst of QuantumLeap-Deloitte and during an activity I had some problems extracting a IPA file containing Chinese characters, so  I thought to add a support for the unicode encoding.

### CODE REVIEW

**Comment by leonjza** [comment]
Thanks for the PR!

I would prefer it if we did not assume that the target file name was `cp437` encoded. Could we instead update this PR to add a flag that will be used as an indicator that `cp437` should be expected?

**Comment by Fabiano1107** [comment]
Yes, of course, but how would it work if it didn't?

**Comment by leonjza** [comment]
Im sorry I don't understand your question. Could you rephrase?

**Comment by Fabiano1107** [comment]
If you just use a flag to indicate that `cp437`  should be expected, what happen if the file is not `cp437` encoded?

**Comment by leonjza** [comment]
The same error you got when originally opening this PR, I think.

**Comment by Fabiano1107** [comment]
So, what you think if I add a check for the file encoding and if the file is not `cp437` encoded the unzip proceeds as in the PR?
Otherwise we proceed as you proposed.

**Comment by leonjza** [comment]
Lets just add a warning stating that you may want the new flag.

**Comment by Fabiano1107** [comment]
Of course, it's perfect.

Thank you for your time.

**Comment by leonjza** [comment]
Just checking in @Fabiano1107, are you going to update the PR to include the changes?

**Comment by Fabiano1107** [comment]
Excuse me for the late response, I'm really sorry.

Yes sure, I will do asap.

**Comment by Fabiano1107** [comment]
Hi @leonjza, I updated the PR with the flag required, but I don't understand the error in `continuous-integration/travis-ci/pr`. On my machine it works without problems.

**Comment by leonjza** [comment]
It failed because the test does not know about the new argument you added. Don't worry about it, I can fix that. Thanks!

**Comment by Fabiano1107** [comment]
Thanks to you, i hope PR is okay.
